awscli/customizations/s3/subcommands.py

```diff
@@ -750,12 +750,35 @@ class SyncCommand(S3TransferCommand):
                 [METADATA, METADATA_DIRECTIVE]
 
 
-class MbCommand(S3TransferCommand):
+class MbCommand(S3Command):
     NAME = 'mb'
     DESCRIPTION = "Creates an S3 bucket."
     USAGE = "<S3Uri>"
-    ARG_TABLE = [{'name': 'paths', 'nargs': 1, 'positional_arg': True,
-                  'synopsis': USAGE}]
+    ARG_TABLE = [{'name': 'path', 'positional_arg': True, 'synopsis': USAGE}]
+
+    def _run_main(self, parsed_args, parsed_globals):
+        super(MbCommand, self)._run_main(parsed_args, parsed_globals)
+
+        path = parsed_args.path
+        if path.startswith('s3://'):
+            path = path[5:]
+        bucket, _ = find_bucket_key(path)
+
+        bucket_config = {'LocationConstraint': self.client.meta.region_name}
+        params = {'Bucket': bucket}
+        if self.client.meta.region_name != 'us-east-1':
+            params['CreateBucketConfiguration'] = bucket_config
+
+        try:
```

Comment:
The logic is essentially:
```
try:
    some_api_call
    return success
except Exception as e:
    print_error(e)
    return failure
```

It's nice to not have to duplicate those lines, but it doesn't seem like that much an issue.  Either way I'd say that the `ResultPrinter` and associated classes is not the right way to handle simple cases like `Mb` and we'd need some other shared abstraction as you suggest.

So the question would be whether or not creating this abstraction should block getting this merged.  I'm ok with adding a comment that we may want to circle back and try to consolidate the code later.